### PR TITLE
OpenMPI: Correct slurm-pmi conflict version

### PR DESCRIPTION
**Pull request by tldahlgren**
The conflict `when` clause doesn't match the conflict message `+pmi is required for openmpi(>=1.5.5) to work with SLURM.`  This allows the package to be installed as `-pmi schedulers=slurm` but it doesn't work for users.

This fix instead generates the conflict:

```
$ spack install openmpi@4.1.2 -pmi schedulers=slurm
==> Error: openmpi@4.1.2~pmi schedulers=slurm is unsatisfiable, errors are:
  A conflict was triggered
  no version satisfies the given constraints

    To see full clingo unsat cores, re-run with `spack --show-cores=full`
    For full, subset-minimal unsat cores, re-run with `spack --show-cores=minimized
    Warning: This may take (up to) hours for some specs
25.492u 1.509s 0:28.37 95.1%	0+0k 1160+1184io 0pf+0w
```

### CODE REVIEW

**Review by alalazo — APPROVED** [review]
Waiting for a maintainer to have a look at this before merging

**Comment by hppritcha** [comment]
I need to test if this pr breaks building master branch.

**Review by hppritcha — APPROVED** [review]

**Comment by hppritcha** [comment]
well master branch will not build with the pmi variant, but since this PR works with release branches its okay.

**Comment by aweits** [comment]
> The conflict `when` clause doesn't match the conflict message `+pmi is required for openmpi(>=1.5.5) to work with SLURM.` This allows the package to be installed as `-pmi schedulers=slurm` but it doesn't work for users.

Hm, this broke our openmpi 4.0.5 builds (well, spack decides to compile mpich instead to work around the conflict which is *extremely* nonintuitive) - we build with ^slurm ~pmi and just use pmix - it works just fine, assuming you've compiled slurm correctly?

**Comment by hppritcha** [comment]
@aweits are you using srun to launch jobs built against your spack install of Open MPI using ^slurm ~pmi?  just trying to understand your use case better.

**Comment by aweits** [comment]
yes - openmpi is built with "fabrics=ucx schedulers=slurm +cuda +thread_multiple +cxx +cxx_exceptions +pmix", slurm is external for us.

**Comment by aweits** [comment]
also see #22086

**Pull request merged**
